tasks/main.yml

```diff
@@ -89,8 +103,12 @@
           {%         if not host2 in __vpn_psks[__vpn_idx] %}
           {%           set _ = __vpn_psks[__vpn_idx].__setitem__(host2, {}) %}
           {%         endif %}
-          {%         set psk = lookup('lines', 'openssl rand -base64 48') %}
-          {%         set val = {'pre_shared_key':psk} %}
+          {%         if 'shared_key_content' in tunnel.hosts[host2] %}
+          {%           set val = {'pre_shared_key':tunnel.hosts[host2]['shared_key_content']} %}
+          {%         else %}
+          {%           set psk = lookup('lines', 'openssl rand -base64 48') %}
+          {%           set val = {'pre_shared_key':psk} %}
+          {%-        endif -%}
```

Comment:
I'm not sure if this will work for us - we have to support EL7 controllers, which means jinja 2.7, which means https://jinja.palletsprojects.com/en/2.9.x/templates/#assignments
```
Scoping Behavior

Please keep in mind that it is not possible to set variables inside a block and have them show up outside of it. This also applies to loops. The only exception to that rule are if statements which do not introduce a scope.
```
I'm not sure if the code qualifies as an "if statement" according to the docs.  We do have CI testing using ansible 2.9 + jinja 2.7, but unfortunately this is broken due to the openssl issue.  Also, these docs apply to jinja 2.9, and jinja 2.7 might be different.  

Let me see what I can do in the meantime

Reply:
I just tested on a RHEL 7.9 host (`python-jinja2-2.7.2-4.el7.noarch`), letting the role generate the PSK by itself, and it worked fine.

Reply:
ah, thanks - I can't even install jinja 2.7 in a venv on my fedora 35 machine